sympy/printing/llvmjitcode.py

```diff
@@ -0,0 +1,192 @@
+from __future__ import print_function, division
+
+'''
+Use llvmlite to create executable functions from Sympy expressions
+
+This module requires llvmlite (https://github.com/numba/llvmlite).
+'''
+
+import ctypes
+
+from sympy.external import import_module
+from sympy.printing.printer import Printer
+from sympy import S
+from sympy.utilities.decorator import doctest_depends_on
+
+llvmlite = import_module('llvmlite')
+if llvmlite:
+    ll = import_module('llvmlite.ir').ir
+    llvm = import_module('llvmlite.binding').binding
+    llvm.initialize()
+    llvm.initialize_native_target()
+    llvm.initialize_native_asmprinter()
+
+
+class LLVMJitPrinter(Printer):
```

Comment:
The theano printer has a `doprint` method. See:

```
Python 3.5.1 |Continuum Analytics, Inc.| (default, Dec  7 2015, 11:16:01) 
Type "copyright", "credits" or "license" for more information.

IPython 4.0.1 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

In [1]: from sympy import *

In [2]: from sympy.abc import *

In [3]: from sympy.printing.theanocode import the
theano           theano_code      theano_function  

In [3]: from sympy.printing.theanocode import TheanoPrinter

In [4]: p = TheanoPrinter()

In [5]: p.doprint(a**2 + b**2)
Out[5]: Elemwise{add,no_inplace}.0
```

But maybe LLVM doesn't have a string rep.